x/programs/rust/wasmlanche-sdk/src/program.rs

```diff
@@ -22,7 +23,16 @@ pub enum ExternalCallError {
     OutOfFuel = 2,
 }
 
-/// Represents the current Program in the context of the caller, or an external
+#[derive(Error, Debug, BorshDeserialize)]
+#[repr(u8)]
+#[non_exhaustive]
+#[borsh(use_discriminant = true)]
+pub enum DeployError {
+    #[error("an error happened during the deploy")]
+    DeployFailure = 0,
+}
```

Comment:
When would we want to return this? Can talk about it in this thread and if we agree that *we do* want to return it, we should add the docs. 

We should probably either just panic here or give details that one would want to act on.

Reply:
yeah, lets worry about this later.